packages/babel-traverse/test/scope.js

```diff
@@ -1038,5 +1038,18 @@ describe("scope", () => {
         },
       });
     });
+    it("shorthand of ObjectProperty inside ObjectPattern", () => {
+      const program = getPath(`
+        const {a} = b;
+      `);
+      program.traverse({
+        VariableDeclaration(path) {
+          path.scope.rename("a", "zzz");
+          expect(
+            path.node.declarations[0].id.properties[0].shorthand,
+          ).toBeFalsy();
+        },
+      });
```

Comment:
Could:
- Move the assertion from inside the visitor to after the `program.traverse` call, to be 100% sure that it runs (I know you copied this pattern from the test above, but we can do better 🙂)
- Replace `.toBeFalsy()` with `.toBe(false)` (since we expect false and not any falsy value)
- Add an `expect(program.toString()).toMatchInlineSnapshot()` at the end, which helps when reading the tests to more easily see what's happening

Also, it would be good to have the following tests:
- `const a = 1, obj = { a };`, renaming `a`
- `const { a } = 1; { const { b } = 2 }` renaming `a` to `b`

Reply:
I did another fix locally.
I don't think it's necessary to traverse twice to fix this, just add the `ObjectProperty` visitor.
At the same time I also found a strange thing, we also have an `extra.shorthand` in the AST, I don't understand why.🤔